internal/jobtracker/jobtracker_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2025 Canonical.
+package jobtracker_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/frankban/quicktest/qtsuite"
+
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/jobtracker"
+	"github.com/canonical/jimm/v3/internal/testutils/jimmtest"
+)
+
+type jobTrackerSuite struct {
+	db      *db.Database
+	tracker *jobtracker.Tracker
+}
+
+func (s *jobTrackerSuite) Init(c *qt.C) {
+	db := &db.Database{
+		DB: jimmtest.PostgresDB(c, time.Now),
+	}
+	err := db.Migrate(context.Background())
+	c.Assert(err, qt.IsNil)
+
+	s.db = db
+	tracker, err := jobtracker.NewJobTracker(db, time.Second*5)
+	c.Assert(err, qt.IsNil)
+	s.tracker = tracker
+}
+
+// !!! --- For PR reviewers, currently we're polling the job status as we are waiting
+// for the forselect to update. We could use gomock / some mock and capture the call. What would you prefer?
+// Gomock would allow us to see the order of calls and expected params, so perhaps a bit cleaner.
+//
+// Or we leave it polling.
+func (s *jobTrackerSuite) TestJobTracker_Run_JobIsSetFailed_JobError(c *qt.C) {
```

Comment:
It's a bit hard to read the test name/intent. What about
```suggestion
func (s *jobTrackerSuite) TestJobTracker_JobWithError(c *qt.C) {
```
Similar sentiment for the other test names.
I think you should even drop the `TestJobTracker` prefix since they're already in the jobTracker suite.

Reply:
I was kind of just name spacing the tests, but sure will go update names.